src/test/kotlin/org/jmailen/gradle/kotlinter/functional/ExtensionTest.kt

```diff
@@ -119,4 +119,66 @@ internal class ExtensionTest : WithGradleTest.Kotlin() {
             assertEquals(TaskOutcome.SUCCESS, task(":lintKotlinMain")?.outcome)
         }
     }
+
+    @Test
+    fun `extension properties are evaluated only during task execution`() {
+        projectRoot.resolve("build.gradle") {
+            @Language("groovy")
+            val buildScript = """
+                plugins {
+                    id 'kotlin'
+                    id 'org.jmailen.kotlinter'
+                }
+                
+                tasks.whenTaskAdded {
+                    // configure all tasks eagerly
+                }
+                
+                kotlinter {
+                    disabledRules = ["filename"]
+                }
+                
+                """.trimIndent()
+            writeText(buildScript)
+        }
+        projectRoot.resolve("src/main/kotlin/FileName.kt") {
+            writeText(kotlinClass("DifferentClassName"))
+        }
+
+        build("lintKotlin").apply {
+            assertEquals(TaskOutcome.SUCCESS, task(":lintKotlinMain")?.outcome)
+        }
+    }
+
+    @Test
+    fun `user customized values take precedence over extension values`() {
+        projectRoot.resolve("src/main/kotlin/FileName.kt") {
+            @Language("kotlin")
+            val kotlinClass = """
+                class Precedence {
+                    fun hi() = Unit
+                }
+            """.trimIndent()
+            writeText(kotlinClass)
+        }
+        projectRoot.resolve("build.gradle") {
+            @Language("groovy")
+            val script = """
+                kotlinter {
+                    disabledRules = ['filename']
+                }
+                
+                lintKotlinMain {
+                    disabledRules = ['final-newline']
+                }
```

Comment:
https://github.com/jeremymailen/kotlinter-gradle/pull/131/files#r376768433

✅ Done
I've added simple case for `disabledRules` only, but if needed I can add more detailed checks for every property, if needed.